posts/how-to-commit-code-to-a-git-repository-with-visual-studio-code.md

```diff
@@ -0,0 +1,172 @@
+
+<figure>
+    <img src="https://storage.googleapis.com/sourcegraph-assets/learn/tutorial-images/vs-code-diff-view.png" alt="An example of a diff view of the changes made to README.md." />
+    <figcaption>An example of a diff view of the changes made to <code>README.md</code>.</figcaption>
+</figure>
+
+To prepare a commit, we will select changes to add to the staging area.
+
+## Step 2 — Adding changes to the staging area
+
+The **staging area** in Git is the collection of changes that will be included in the next commit. The next step is to add some changes to the staging area.
+
+### Staging individual files
+
+Each file in the **Changes** list represents a change in the repository’s working directory that has not yet been staged and is available for staging.
+
+The badge next to each file indicates the type of change:
+
+- The `M` badge indicates that a file is modified. This means that it exists in the repository and has changed since the last commit.
+- The `U` badge represents an untracked file, which means it doesn’t exist in the repository and was newly created since the last commit.
+- The `D` badge means that the file was deleted.
+
+Each file has three action buttons. You can reveal these buttons by hovering or clicking on a file in the list.
+
+<figure className="small" >
+    <img src="https://storage.googleapis.com/sourcegraph-assets/learn/tutorial-images/vs-code-source-control-stage-file.png" alt="Actions for each file in the Changes section." />
+    <figcaption>Actions for each file in the <strong>Changes</strong> section.</figcaption>
+</figure>
+
+From left to right, the action buttons are:
+
+- **Open File**, which opens the file in a new editor tab for editing.
+- **Discard Changes**. This will discard the changes to this file, reverting it to the version from the latest commit.
+- **Stage Changes**. This will add the file to the staging area for inclusion in the next commit.
+
+To add a file to the staging area, click on its **Stage Changes** icon, represented by a `+` sign.
+
+For example, we can stage the `README.md` file in the above screenshot by clicking on it in the list to reveal the action buttons and then selecting the `+` icon.
+
+If we want to add more changes, we can repeat this process for other files. For example, we can continue by staging the untracked file, `getting-started.txt`. We can add as many changes to a commit as we want.
+
+### Staging all files
+
+In situations where we have many changed files and we want to stage them all, we can use the **Stage All Changes** button. This button is revealed by hovering over the title of the **Changes** section. Its icon is a `+` sign, as illustrated in the screenshot below.
+
+<figure className="small">
```

Comment:
Make image markdown. This image and a number of the others may benefit from more context (showing a little bit around them)